trin-core/src/types/accumulator.rs

```diff
@@ -0,0 +1,203 @@
+use ethereum_types::U256;
+use ssz::Encode;
+use ssz_derive::{Decode, Encode};
+use ssz_types::{typenum, VariableList};
+use tree_hash::{MerkleHasher, TreeHash};
+use tree_hash_derive::TreeHash;
+
+use crate::types::header::Header;
+
+const EPOCH_SIZE: usize = 8192; // number of blocks / epoch
+const MAX_HISTORICAL_EPOCHS: usize = 131072; // 2 ** 17
+
+// ssz list[Hash256, max_length = MAX_HISTORICAL_EPOCHS]
+pub type HistoricalEpochsList = VariableList<tree_hash::Hash256, typenum::U8192>;
+
+// ssz list[HeaderRecord, max_length = EPOCH_SIZE]
+pub type HeaderRecordList = VariableList<HeaderRecord, typenum::U131072>;
+
+// ssz container
```

Comment:
This is not a doc comment and doesn't give us any information on what `MasterAccumulator` is and what it is doing.

Reply:
This one I slightly disagree. I find it somewhat useful as I'm putting together the accumulator logic to remember that this object is an ssz-encoded container, not a list.